airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/operation/OperationType.kt

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.operation
+
+/**
+ * Defines the operations that may be invoked via the CLI arguments.  Not all connectors will
+ * implement all of these operations.
+ */
+enum class OperationType {
```

Comment:
I'm not a big fan of defining an enum when we know no connector will provide all of the operations

Reply:
This is really just a port of an existing enum in the CDK.  We can definitely change this/make this better as we go.